core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -495,6 +495,17 @@ class CivilizationInfo {
         updateStatsForNextTurn()
     }
 
+    fun goldCostOfSignResearchAgreement(): Int {
+        // https://forums.civfanatics.com/resources/research-agreements-bnw.25568/
+        return (when (getEra()) {
```

Comment:
Split this into a = when{...}
return a*(...getModifier())

Reply:
i will change it, but i think it has better performance.